arch/risc-v/src/common/riscv_exception_common.S

```diff
@@ -83,20 +83,6 @@ exception_common:
   addi       sp, sp, -XCPTCONTEXT_SIZE
   save_ctx   sp
 
-  csrr       s0, CSR_STATUS
-  REGSTORE   s0, REG_INT_CTX(sp)  /* status */
-
-  addi       s0, sp, XCPTCONTEXT_SIZE
-  REGSTORE   s0, REG_X2(sp)       /* original SP */
-
-  csrr       s0, CSR_EPC
-  REGSTORE   s0, REG_EPC(sp)      /* exception PC */
-
-#ifdef CONFIG_ARCH_FPU
-  mv         a0, sp
-  jal        x1, riscv_savefpu    /* save FPU context */
```

Comment:
Another option would be to define riscv_savefpu as a macro.

Reply:
Now riscv_savefpu/riscv_loadfpu defined as macro